### PR TITLE
[prometheus-pingmesh-exporter] Add Prometheus Pingmesh Exporter

### PR DESCRIPTION
**Pull request by dongjiang1989**
Pingmesh : A Large-Scale System for Data Center Network Latency. [pingmesh-agent](https://github.com/kubeservice-stack/pingmesh-agent) is prometheus exporter

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-pingmesh-exporter]`)

### CODE REVIEW

File: charts/prometheus-pingmesh-exporter/values.yaml [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
+  ##
+  # pullSecrets:
+  #   - myRegistrKeySecretName
+
+podSecurityContext: {}
+# fsGroup: 1000
+
+## User and Group to run pingmesh-exporter container as
+securityContext:
```
Comment (monotek) [inline charts/prometheus-pingmesh-exporter/values.yaml:89]: is this a privilege contaienr?

if not please add:

```
            privileged: false

```

please also add:

```
            seccompProfile:
              type: RuntimeDefault
```

sorting values alphabetically woudl also be nice

File: charts/prometheus-pingmesh-exporter/values.yaml [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+  runAsGroup: 1000
+  readOnlyRootFilesystem: true
+  runAsNonRoot: true
+  allowPrivilegeEscalation: false
+  capabilities:
+    drop: ["ALL"]
+# Add NET_RAW to enable ICMP
+#    add: ["NET_RAW"]
+
+livenessProbe:
```
Comment (monotek) [inline charts/prometheus-pingmesh-exporter/values.yaml:100]: liveness and readinessprobe should not be the same

File: charts/prometheus-pingmesh-exporter/values.yaml [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+    interval: 60.0
+    delay: 200
+    timeout: 2.0
+    source_ip_addr: 0.0.0.0
+    ip_protocol: ip6
+  mesh:
+    add-ping-public: 
+      name: ping-public-demo
+      type: OtherIP
+      ips :
```
Comment (monotek) [inline charts/prometheus-pingmesh-exporter/values.yaml:217]: can we please remove the chinese urls and ips here?

File: charts/prometheus-pingmesh-exporter/values.yaml [hunk 4]
```diff
@@ -0,0 +1,396 @@
...
+## Additional secret mounts
+# Defines additional mounts with secrets. Secrets must be manually created in the namespace.
+extraSecretMounts: []
+  # - name: secret-files
+  #   mountPath: /etc/secrets
+  #   secretName: pingmesh-secret-files
+  #   readOnly: true
+  #   defaultMode: 420
+
+resources:
```
Comment (monotek) [inline charts/prometheus-pingmesh-exporter/values.yaml:244]: request and limits should not be set by default but be commented out

File: charts/prometheus-pingmesh-exporter/values.yaml [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+  ##
+  selfMonitor:
+    enabled: true
+    additionalMetricsRelabels: {}
+    additionalRelabeling: []
+    labels: {}
+    interval: 30s
+    scrapeTimeout: 30s
+
+  ## If true, a ServiceMonitor CRD is created for a prometheus operator
```
Comment (monotek) [inline charts/prometheus-pingmesh-exporter/values.yaml:327]: please don't add crds by default.
might be also just wrong comment and a CR is meaned?

File: charts/prometheus-pingmesh-exporter/values.yaml [hunk 6]
```diff
@@ -0,0 +1,396 @@
...
+    module: dns
+  ## scheme: HTTP scheme to use for scraping. Can be used with `tlsConfig` for example if using istio mTLS.
+  scheme: http
+  ## tlsConfig: TLS configuration to use when scraping the endpoint. For example if using istio mTLS.
+  ## Of type: https://github.com/coreos/prometheus-operator/blob/master/Documentation/api.md#tlsconfig
+  tlsConfig: {}
+  bearerTokenFile:
+
+  targets:
+    - name: baidu.com                 # Human readable URL that will appear in Prometheus / AlertManager
```
Comment (monotek) [inline charts/prometheus-pingmesh-exporter/values.yaml:348]: please remove chinese domain

File: charts/prometheus-pingmesh-exporter/values.yaml [hunk 7]
```diff
@@ -0,0 +1,396 @@
...
+      additionalRelabeling: []         # List of metric relabeling actions to run
+
+## Custom PrometheusRules to be defined
+## ref: https://github.com/coreos/prometheus-operator#customresourcedefinitions
+prometheusRule:
+  enabled: false
+  additionalLabels: {}
+  namespace: ""
+  rules:
+  - alert: 站点可用性
```
Comment (monotek) [inline charts/prometheus-pingmesh-exporter/values.yaml:372]: example should be in english

File: charts/prometheus-pingmesh-exporter/Chart.yaml [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+- pimgmesh
+- monitoring
+maintainers:
+- email: dongjiang1989@126.com
+  name: dongjiang
+name: prometheus-pingmesh-exporter
+sources:
+- https://github.com/kubeservice-stack/pingmesh-agent
+type: application
+version: 1.0.1
```
[the remark this answers is not in the document]
Comment (dongjiang1989) [reply charts/prometheus-pingmesh-exporter/Chart.yaml]: Thinks for you reviews! Done, change chart version to `0.1.0`

[the remark this answers is not in the document]
Comment (dongjiang1989) [reply charts/prometheus-pingmesh-exporter/Chart.yaml]: Thinks for you reviews! Done, change chart version to `0.1.0`

File: charts/prometheus-pingmesh-exporter/templates/podsecuritypolicy.yaml [hunk 9]
```diff
@@ -0,0 +1,41 @@
+{{- if and .Values.pspEnabled (.Capabilities.APIVersions.Has "policy/v1beta1/PodSecurityPolicy") }}
+apiVersion: policy/v1beta1
+kind: PodSecurityPolicy
```
[the remark this answers is not in the document]
Comment (dongjiang1989) [reply charts/prometheus-pingmesh-exporter/templates/podsecuritypolicy.yaml]: Done! remove `PodSecurityPolicy`

File: charts/prometheus-pingmesh-exporter/templates/role.yaml [hunk 10]
```diff
@@ -0,0 +1,18 @@
+{{- if .Values.pspEnabled }}
```
[the remark this answers is not in the document]
Comment (dongjiang1989) [reply charts/prometheus-pingmesh-exporter/templates/role.yaml]: Done. remove `pspEnabled` config

File: charts/prometheus-pingmesh-exporter/values.yaml (hunk 1 above)
Comment (dongjiang1989) [reply charts/prometheus-pingmesh-exporter/values.yaml:90]: Done. Add `privileged` and `seccompProfile`

File: charts/prometheus-pingmesh-exporter/values.yaml (hunk 2 above)
Comment (dongjiang1989) [reply charts/prometheus-pingmesh-exporter/values.yaml:104]: change `readiness` probe to `/ready` handler

File: charts/prometheus-pingmesh-exporter/values.yaml (hunk 7 above)
Comment (dongjiang1989) [reply charts/prometheus-pingmesh-exporter/values.yaml]: Done. change example with english name

File: charts/prometheus-pingmesh-exporter/values.yaml (hunk 6 above)
Comment (dongjiang1989) [reply charts/prometheus-pingmesh-exporter/values.yaml]: change chinese domain `baidu.com` to `google.com`

File: charts/prometheus-pingmesh-exporter/values.yaml (hunk 5 above)
Comment (dongjiang1989) [reply charts/prometheus-pingmesh-exporter/values.yaml:325]: Done. default value set `false`

File: charts/prometheus-pingmesh-exporter/values.yaml (hunk 4 above)
Comment (dongjiang1989) [reply charts/prometheus-pingmesh-exporter/values.yaml]: Done. Setting resources is `{}` and be commented out

File: charts/prometheus-pingmesh-exporter/values.yaml (hunk 3 above)
Comment (dongjiang1989) [reply charts/prometheus-pingmesh-exporter/values.yaml:221]: Done. Remove the chinese urls and ip

File: charts/prometheus-pingmesh-exporter/templates/selfservicemonitor.yaml [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+    {{- if .Values.serviceMonitor.selfMonitor.labels  }}
+    {{- toYaml (.Values.serviceMonitor.selfMonitor.labels) | nindent 4 }}
+    {{- end }}
+spec:
+  endpoints:
+  - path: /metrics
+    interval: {{ .Values.serviceMonitor.selfMonitor.interval }}
+    scrapeTimeout: {{ .Values.serviceMonitor.selfMonitor.scrapeTimeout }}
+    scheme: http
+
```
[the remark this answers is not in the document]
Comment (dongjiang1989) [reply charts/prometheus-pingmesh-exporter/templates/selfservicemonitor.yaml:19]: Done

File: charts/prometheus-pingmesh-exporter/values.yaml [hunk 12]
```diff
@@ -0,0 +1,396 @@
...
+  #         memory: 16Mi
+  #       requests:
+  #         memory: 4Mi
+  #         cpu: 20m
+  #   volumeMounts:
+  #     - mountPath: /etc/prometheus/secrets/pingmesh-tls
+  #       name: secret-pingmesh-tls
+
+## Enable pod security policy
+pspEnabled: true
```
[the remark this answers is not in the document]
Comment (dongjiang1989) [reply charts/prometheus-pingmesh-exporter/values.yaml]: Done. remove values config

**Comment by dongjiang1989** [comment]
@monotek  Please Take A Look.

**Review by monotek — CHANGES_REQUESTED** [review]
please fix ci

**Comment by dongjiang1989** [comment]
> please fix ci

Thinks for review!  Fix CI pipeline Done. @monotek

**Review by zanhsieh — CHANGES_REQUESTED** [review]
@dongjiang1989 
Thank you for contribution. Would you sign DCO please?

**Comment by dongjiang1989** [comment]
> @dongjiang1989 Thank you for contribution. Would you sign DCO please?

Thinks for reivew! Sign DCO Done. @zanhsieh

**Review by monotek — APPROVED** [review]

**Pull request merged**